addons/sale/models/sale_order.py

```diff
@@ -70,7 +70,7 @@ def _rec_names_search(self):
         required=True, readonly=False, change_default=True, index=True,
         tracking=1,
         states=READONLY_FIELD_STATES,
-        domain="[('company_id', 'in', (False, company_id))]")
+        domain="['|', ('company_id', 'parent_of', company_id), ('company_id', '=', False)]")
```

Comment:
Only on SO? not on PO or other models ?
Also, is the `check_company` adapted to the logic you are deciding and enforcing ?  If so, might be cleaner and easier to use `check_company=True` here to enforce one single logic everywhere.

Reply:
Yes, just the `check_company` is enough and safer regarding consistency